pytential/symbolic/execution.py

```diff
@@ -1030,6 +1033,15 @@ def eval(self, context=None, timing_data=None,
         if context is None:
             context = {}
 
+        # NOTE: avoid compiling any code if the expression is long lived
+        # and already nicely cached in the collection from a previous run
+        import pymbolic.primitives as prim
+        if isinstance(self.sym_op_expr, prim.CommonSubexpression) \
+                and self.sym_op_expr.scope == sym.cse_scope.DISCRETIZATION:
+            cache = self.places._get_cache("cse")
```

Comment:
Use a specially-made class symbol here instead of "cse", to make clashes less likely.

Reply:
That's what's used in [EvaluationMapper.map_common_subexpression](https://github.com/inducer/pytential/blob/d3b25669c95335b515b7b5b97833f75a70f19b5b/pytential/symbolic/execution.py#L260-L274). Should I change it there too?

Reply:
Thanks for noticing this. Yeah, I think so. I think that code was written before the message "symbols are unique-er than strings but just as hashable" sunk in with me.

Reply:
Went ahead and updated this and other cache names I could find in 38a6290.

Reply:
Pushed a counterproposal in 365a591. Let me know if you're OK with that.

Reply:
Yep, looks good to me!